runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -547,6 +552,22 @@ ClientSessionData::clearCaches()
       }
 
    _wellKnownClasses.clear();
+   setCachesAreCleared(true);
+   }
+
+void
+ClientSessionData::clearCachesLocked(TR_J9VMBase *fe)
+   {
+   // Acquire all required locks to clear client session caches
+   // This should be used when caches need to be cleared
+   // while the client is still sending requests
+   getSequencingMonitor()->enter();
```

Comment:
if we execute `clearCachesLocked()` before releasing the sequencing monitor, then we don't have to acquire it here again.

Reply:
Yeah, but I acquire the lock inside `clearCachesLocked` in case that method ever gets called in another place where we do not have this monitor.